cmd/kubeadm/app/preflight/checks.go

```diff
@@ -431,9 +431,12 @@ func (hst HTTPProxyCheck) Name() string {
 // Check validates http connectivity type, direct or via proxy.
 func (hst HTTPProxyCheck) Check() (warnings, errorList []error) {
 	klog.V(1).Infoln("validating if the connectivity type is via proxy or direct")
-	u := (&url.URL{Scheme: hst.Proto, Host: hst.Host}).String()
+	u := &url.URL{Scheme: hst.Proto, Host: hst.Host}
+	if utilsnet.IsIPv6String(hst.Host) {
+		u.Host = net.JoinHostPort(hst.Host, "1234")
```

Comment:
May I know where "1234" came from ?

Reply:
It is random port to assign in the formed URL. For detecting proxy configuration it is ignored, so it wouldn't be any difference with "80", "443" or any other port valid URL port number.

Reply:
After reading the code, I got same conclusion.